src/assets/Generator.Shared/RawRequestUriBuilder.cs

```diff
@@ -87,5 +90,56 @@ private enum RawWritingPosition
             Port,
             Rest
         }
+
+        public void AppendRawNextLink(string nextLink, bool escape)
+        {
+            (string Name, string Value) GetQueryParts(string queryUnparsed)
+            {
+                int separatorIndex = nextLink.IndexOf(QueryValueSeparator);
+                return (queryUnparsed.Substring(0, separatorIndex), queryUnparsed.Substring(separatorIndex + 1));
+            }
+
+            RawNextLinkWritingPosition position = RawNextLinkWritingPosition.Path;
```

Comment:
This logic should go into AppendRaw

Reply:
Okay. That'll take me some time. If I'm going to modify it, I'd like to have some tests around it.

Reply:
There is a test class for it https://github.com/Azure/autorest.csharp/blob/feature/v3/test/AutoRest.Shared.Tests/RawRequestUriBuilderTest.cs

Reply:
Done. This logic is now in AppendRaw. Added several test cases and fixed an issue when using AppendRaw to do paths after a port was specified.